src/java/com/unifina/signalpath/ModuleExceptionMessage.java

```diff
@@ -2,12 +2,16 @@
 
 import java.util.Map;
 
-public class ModuleExceptionMessage {
-	public ModuleExceptionMessage(int hash, Map<String, Object> msg) {
+public abstract class ModuleExceptionMessage {
+	protected int hash;
+
+	ModuleExceptionMessage(int hash) {
 		super();
 		this.hash = hash;
-		this.msg = msg;
 	}
-	private int hash;
-	private Map<String,Object> msg;
+
+	/**
+	 * For front-end JSON output.
+	 */
+	public abstract Map<String, Object> toMap();
```

Comment:
Writing the module id could happen on this layer. "hash" is a bad name, let's call it "module".
```suggestion
	public Map<String, Object> toMap() {
	    Map<String, Object> map = new HashMap<>();
	    map.put("module", hash);
	    return map;
	}
```